opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java

```diff
@@ -596,7 +596,7 @@ public void populateDocument(Document doc, File file, String path, AbstractAnaly
         doc.add(new SortedDocValuesField(QueryBuilder.FULLPATH,
                 new BytesRef(file.getAbsolutePath())));
 
-        if (RuntimeEnvironment.getInstance().isHistoryEnabled()) {
+        if (HistoryGuru.getInstance().repositorySupportsHistory(file)) {
```

Comment:
uses `HistoryGuru#getRepository(file)` which is based on cached repository lookup so this should not incur performance penalty.